framework/src/modules/fee/module.ts

```diff
@@ -127,6 +127,15 @@ export class FeeModule extends BaseInteroperableModule {
 
 	public async beforeCommandExecute(context: TransactionExecuteContext): Promise<void> {
 		const { transaction, header } = context;
+
+		// Same check as in verifyTransaction()
+		// Required to prevent negative available fee when doing a dry run in non-strict mode
+		const minFee = this._getMinFee(header.height, transaction.getBytes().length);
+		const availableFee = transaction.fee - minFee;
```

Comment:
implementation wise this is bigint so it's ok, but type-wise both are `uint64` so that `availableFee` will not be in the range of `uint64`.
In the typed system it would be better to compare before subtraction